go-controller/pkg/ovn/master.go

```diff
@@ -449,14 +458,20 @@ func (oc *Controller) syncGatewayLogicalNetwork(node *kapi.Node, l3GatewayConfig
 		if err != nil {
 			return err
 		}
-		mpIP := util.GetNodeManagementIfAddr(subnets[0]).IP.String()
-
-		if err := addPolicyBasedRoutes(node.Name, mpIP, l3GatewayConfig.IPAddresses); err != nil {
-			return err
-		}
+		for _, subnet := range subnets {
+			hostIfAddr := util.GetNodeManagementIfAddr(subnet)
+			for _, l3GatewayConfigIP := range l3GatewayConfig.IPAddresses {
+				if utilnet.IsIPv6(hostIfAddr.IP) == utilnet.IsIPv6(l3GatewayConfigIP.IP) {
```

Comment:
Rather than having 2 nested loops, you should be able to loop over only `subnets`, and use `util.MatchIPFamily` to find the `l3GatewayConfig.IPAddresses` value that goes with each subnet.